### ISSUE TITLE
Closing manager window minimizes it instead (Linux)

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1348]
**Reported by Nicolas on 4 Apr 45021490 12:25 UTC**
On Linux, clicking the Close button on the Manager titlebar
minimizes the window.
Not to the tray, but to the normal task bar.
In other words, the Close button does
exactly the same as the Minimize button,
making it pretty useless.

FileClose Window also minimizes the manager.
I couldn't find any way to minimize to tray.

My desktop environment is KDE Plasma 4.11.13.

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1377

**Comment by shalnoff** [comment]
agree. Moreover, the window doesn't minimise to tray at all (Mate, Linux Mint 32bit. BOINIC 7.0.27)
would be great to fix it. 

thank you for great applicaiton!

**Comment by Songworks** [comment]
Any plans to fix this?
I like the tray icons Snooze feature. But the icon is obviously only present if the managers window is also open.
System: Linux Mint 17.3 running boinc 7.6.2 64bit.

**Comment by eMPee584** [comment]
Yes, please make BOINC Manager minimize to tray. I like to keep it running for the "snooze" option, but it wastes space in the task list.

**Comment by eMPee584** [comment]
Actually, found a way to snooze BOINC from the CLI (f.e. for 15 minutes):
`boinccmd --set_run_mode never $((60*15))`
This will probably obliterate the GUI for me.

**Comment by elfring** [comment]
Can this special software behaviour happen because the support for the functionality “Minimize window into system tray” is unclear for [the usage of the member function “wxTopLevelWindow::Iconize”](http://docs.wxwidgets.org/trunk/classwx_top_level_window.html#ae461559192e33ef111c3f147999a26bf "Clarification for usage of “wxTopLevelWindow::Iconize”?") which is called by [the member function “CBOINCBaseFrame::OnClose”](https://github.com/BOINC/boinc/blob/e6607a63402a2b4ce2d88588efa2c011ffae9b86/clientgui/BOINCBaseFrame.cpp#L325 "Closing an application window") so far?

**Comment by ChristianBeer** [comment]
Problem still present with 7.6.33 on Debian 9 (KDE)

**Comment by Ageless93** [comment]
@lfield, one for your list of fixes?

**Comment by lfield** [comment]
It is not on my list and has not been tested on 7.8.3. Does anyone have KDE and can test? On my Ubuntu machine close does seem to minimize rather than exit the application so it may be a general problem. I thought that this was a 'feature' rather than a bug.

**Comment by Germano0** [comment]
Before filling a pull request I would like to hear some feedbacks.
I elaborated two different patches:

- send to tray icon patch;
- close window patch.

**Send to tray icon patch**
Closes the BOINC manager window, leaving only the tray icon.
Problems: 
1. for unknown reasons, once BOINC manager is reduced to tray icon, if you right click on tray icon and click on "Exit", the icon will disappear, but BOINC manager process will hang;
2. GNOME has no tray icons bar at all, so IMHO having BOINC manager reduced to icon is useless
3. BOINC runs as a service, regardless if the manager is opened / closed
So for the previous points, I would even remove the tray icon at all from BOINC Manager for Linux
```
diff -urNr boinc-client_release-7.14-7.14.2-orig/clientgui/BOINCBaseFrame.cpp boinc-client_release-7.14-7.14.2/clientgui/BOINCBaseFrame.cpp
--- boinc-client_release-7.14-7.14.2-orig/clientgui/BOINCBaseFrame.cpp	2018-10-11 02:43:27.000000000 +0200
+++ boinc-client_release-7.14-7.14.2/clientgui/BOINCBaseFrame.cpp	2018-12-05 14:42:08.350622590 +0100
@@ -329,13 +329,7 @@
         wxGetApp().FrameClosed();
         Destroy();
     } else {
-#ifdef __WXGTK__
-        // Apparently aborting a close event just causes the main window to be displayed
-        // again.  Just minimize the window instead.
-        Iconize();
-#else
         Hide();
-#endif
     }
 
     wxLogTrace(wxT("Function Start/End"), wxT("CBOINCBaseFrame::OnClose - Function End"));

```

**Close window patch**
```
diff -urNr boinc-client_release-7.14-7.14.2-orig/clientgui/BOINCBaseFrame.cpp boinc-client_release-7.14-7.14.2-close/clientgui/BOINCBaseFrame.cpp
--- boinc-client_release-7.14-7.14.2-orig/clientgui/BOINCBaseFrame.cpp	2018-10-11 02:43:27.000000000 +0200
+++ boinc-client_release-7.14-7.14.2-close/clientgui/BOINCBaseFrame.cpp	2018-12-05 15:21:43.728546201 +0100
@@ -330,9 +330,8 @@
         Destroy();
     } else {
 #ifdef __WXGTK__
-        // Apparently aborting a close event just causes the main window to be displayed
-        // again.  Just minimize the window instead.
-        Iconize();
+        wxGetApp().FrameClosed();
+        Destroy();
 #else
         Hide();
 #endif
```

**Comment by JuhaSointusalo** [comment]
People are arguing over at #2604 what exactly the Manager should do when the window is closed.

**Comment by AenBleidd** [comment]
Fixed via #3095

**Comment by coyote3** [comment]
When I "Close Window" on Linux Mint 19.2, "BOINC Manager 7.9.3" still minimizes to taskbar.

Is this because Mint hasn't updated it's BOINC yet?  (If so then I'll just be patient.)(Apologies, I'm a longtime BOINC but a total Linux noob.)
